openghg/store/base/_datasource.py

```diff
@@ -364,10 +372,37 @@ def add_timed_data(
 
         self._last_updated = timestamp_str_now
 
+    def add_hashes(self, version: str, file_hashes: Dict) -> None:
+        """Add hashes of original data files to metadata
+
+        Args:
+            version: Version
+            file_hashes: SHA1 file hash
+        Returns:
+            None
+        """
+        # We'll filepaths in the file_hashes dictionary
+        filename_only = {}
+        for file_hash, filepath in file_hashes.items():
+            try:
+                filename = filepath.name
+            except AttributeError:
+                filename = filepath
+
+            filename_only[file_hash] = filename
+
+        if "original_file_hashes" not in self._metadata:
+            self._metadata["original_file_hashes"] = {version: filename_only}
+        else:
+            if version in self._metadata["original_file_hashes"]:
+                self._metadata["original_file_hashes"][version].update(filename_only)
+            else:
+                self._metadata["original_file_hashes"][version] = filename_only
+
     def delete_all_data(self) -> None:
```

Comment:
When we delete data would we also want to delete the org files and references to the file hashes? I'm just thinking that when we've tried deleting data and then adding again, we seem to be retaining a memory of that data at the moment.

Reply:
Oops! The file hashes and removed from the storage class and the original files are deleted now, added with 8ae5852589e840fb97beec129b48a3c3be6cbb04.

Reply:
@brendan-m-murphy raised the valid point that hashes may be related more than to one datasource so I think it's worth considering what happens if:
1. Datasource and associated hash is removed but hash is related to multiple datasources
2. Try to add back data using original file but this won't add because it finds *some* of the data already present
This can be a separate issue and we could put this in and then deal with that later if needed but it is something we will need to work out how to deal with at some point.

Reply:
Yeah good point, I've opened https://github.com/openghg/openghg/issues/1005 for this.